pkg/kp/consulutil/watch.go

```diff
@@ -229,6 +229,8 @@ func WatchDiff(
 	outCh := make(chan *WatchedChanges)
 
 	go func() {
+		defer close(outCh)
```

Comment:
recommend no squash on those if they are separate changes. each commit should be one logical change